src/api/index.ts

```diff
@@ -77,59 +70,71 @@ export type TravisCIBuildResponse = {
   updated_at: string;
 };
 
+type FetchParams = {
+  limit: number;
+  offset: number;
+  repoSlug: string;
+};
+
 export class TravisCIApi {
-  async retry(
-    travisVersion: string,
-    buildNumber: number,
-    { token }: { token: string },
-  ) {
-    return fetch(`https://api.${travisVersion}/build/${buildNumber}/restart`, {
-      headers: createHeaders(token),
-      method: 'post',
-    });
+  baseUrl: string;
+
+  constructor(baseUrl: string = 'http://localhost:7000') {
+    this.baseUrl = baseUrl;
   }
 
-  async getBuilds(
-    {
-      travisVersion,
-      limit = 10,
-      offset = 0,
-    }: {
-      travisVersion: string;
-      limit: number;
-      offset: number;
-    },
-    { token, owner, repo }: { token: string; owner: string; repo: string },
-  ) {
-    const repoSlug = encodeURIComponent(`${owner}/${repo}`);
+  async retry(buildNumber: number) {
+    return fetch(
+      `${this.baseUrl}${API_BASE_URL}/build/${buildNumber}/restart`,
+      {
+        method: 'post',
+        headers: new Headers({
+          'Travis-API-Version': '3',
+        }),
+      },
+    );
+  }
 
-    const response = await (
-      await fetch(
-        `https://api.${travisVersion}/repo/${repoSlug}/builds?offset=${offset}&limit=${limit}`,
-        {
-          headers: createHeaders(token),
-        },
-      )
-    ).json();
+  async getBuilds({ limit = 10, offset = 0, repoSlug }: FetchParams) {
+    const response = await fetch(
+      `${this.baseUrl}${API_BASE_URL}/repo/${encodeURIComponent(
```

Comment:
Should thus be both this.baseUrl and API_BASE_URL here?

Reply:
In CircleCI backend url and base url are combined at the constructor, I think that I'll do the same here.

Reply:
silly question: no need for an extra `/` in between?

Reply:
That was an outdated "experiment", it would concatenate it to "https://api.travis-ci.com/repo..."